### ISSUE TITLE
Exported variable 'X' has or is using name 'Y' from external module "Z" but cannot be named.

### ISSUE THREAD

**Issue by robyoder** [Microsoft/TypeScript#20657]
**TypeScript Version:**  2.7.0-dev.20171212

**Code**

```ts
// thingA.ts
export interface ThingA { }

// thingB.ts
export interface ThingB { }

// things.ts
export { ThingA } from "./thingA";
export * from "./thingB";

// index.ts
import * as things from "./things";
export const thing1 = (param: things.ThingA) => null;
export const thing2 = (param: things.ThingB) => null;

// tsconfig.json
{
  "compilerOptions": {
    "declaration": true
  }
}
```

**Expected behavior:**

Project compiles without errors.

**Actual behavior:**

```
index.ts(3,14): error TS4023: Exported variable 'thing2' has or is using name 'ThingB' from external module "/Users/rob/Projects/tsctest/thingB" but cannot be named.
```

**Additional information:**

@weswigham fixed the following very similar issues in #19852:

* #5938
* #8612
* #19825

In fact, before his fix, line 2 of index.ts would also fail.

I've pushed this repro as a project here: https://github.com/robyoder/tsctest

**Comment by johnpapa** [comment]
@mhegazy thanks for fixing this .. can you clarify what we need t do to get this fix?

**Comment by mhegazy** [comment]
`typescript@rc` or `typescript@next` should have the fix. the fix should also be included in `typescript@2.7.1` coming out before the end of the month.

**Comment by robyoder** [comment]
Super weird: This issue is not fixed when using a type named `Account`, but it's fixed for everything else. Opened #21313.
